src/internal/pfsdb/commit_provenance.go

```diff
@@ -0,0 +1,275 @@
+package pfsdb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	pfsserver "github.com/pachyderm/pachyderm/v2/src/server/pfs"
+)
+
+// returns the commit of a certain repo in a commit set.
+func ResolveCommitProvenance(ctx context.Context, tx *pachsql.Tx, repo *pfs.Repo, commitSet string) (*pfs.Commit, error) {
```

Comment:
I think we need to make some changes to the context and tx usage. We could either not have a context and rely on the context used when setting up the tx, or ensure that the context we use is derived from the context used when setting up the tx. It seems like the former has been the desired approach up to this point. Also, there is a TODO about removing the ctx field from the tx.